lib/semantics/semantics.cc

```diff
@@ -39,7 +39,9 @@
 
 namespace Fortran::semantics {
 
+using NameToSymbolMap = std::map<const char *, const Symbol *>;
 static void DoDumpSymbols(std::ostream &, const Scope &, int indent = 0);
+static void GetAllSymbols(const Scope &, NameToSymbolMap &);
```

Comment:
Why is a prototype needed for this static function?

Reply:
I thought that it is good practice to have function prototypes so that functions can be referenced before it's defined. I also see that PutIndent has a function prototype in the same way. Can you explain to me why it would be a bad idea to have a function prototype here?

Reply:
In modern C++ usage, prototypes of non-member functions are rare outside of header files unless they're required for mutual recursion.  It's more common to define the static function ahead of its first use, and then you don't have to worry about keeping the prototype and the definition consistent in later changes.

Reply:
I see. Then do you think PutIndent should be defined earlier and its prototype removed?

Reply:
It should probably have been written so, yes.